### PR TITLE
Document the buildVirtualDevice task

### PR DESCRIPTION
**Pull request by denisbedin**
https://docs.microej.com/en/feature-microej-sdk-6-buildvirtualdevice-task/

### CODE REVIEW

**Comment by tdelhomenie-microej** [comment]
@denisbedin The documentation describes correctly how to build a VD, but we don't describe exactly what is a VD, in which case we have to build a VD, ... I think it is too early to start a new page explaining what is a VD since it is quickly changing, but we should at least add a link in an introduction sentence to the current VD page https://docs.microej.com/en/feature-microej-sdk-6-buildvirtualdevice-task/ApplicationDeveloperGuide/virtualDevice.html.
Also we should start a Reference page somewhere in SDK6 section which describes precisely what is the structure of a VD.

File: SDK6UserGuide/appendix/virtualDevice.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+   |   |   |   |- vd-init.xml 
+   |   |- tools
+   |   |- architecture.properties
+   |   |- release.properties
+   |   |- veePort.properties
+   |   |- workbenchExtension*.jar
+
+The Virtual Device contains the Simulation part files of the VEE Port used to build it: 
+
+- the ``javaLibs/`` folder, that contains the Generic Foundation Libraries.
```
Comment (bguedas) [inline SDK6UserGuide/appendix/virtualDevice.rst:41]:
```suggestion
- the ``javaLibs/`` folder, that contains the Foundation Libraries which are common to MICROJVM and S3
```

File: SDK6UserGuide/appendix/virtualDevice.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   |   |- scripts
+   |   |   |- init-vd 
+   |   |   |   |- vd-init.xml 
+   |   |- tools
+   |   |- architecture.properties
+   |   |- release.properties
+   |   |- veePort.properties
+   |   |- workbenchExtension*.jar
+
+The Virtual Device contains the Simulation part files of the VEE Port used to build it:
```
Comment (bguedas) [inline SDK6UserGuide/appendix/virtualDevice.rst:39]: as for JavaDoc, only capitalize the first letter and end with a dot if the item is a proper sentence. See https://www.grammarly.com/blog/bullet-points/

File: SDK6UserGuide/appendix/virtualDevice.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+- the ``MICROJVM/javaLibs/`` folder, that contains the Foundation Libraries specific to the MicroEJ Core Engine.
+
+- the ``MICROJVM/soar/`` folder, that contains the Jar files of the SOAR.
+
+- the ``mocks/`` folder, that contains the Jar files of the mocks for Foundation Libraries.
+
+- the ``resources/`` folder, that contains the OS specific libraries.
+
+- the ``s3/`` folder.
```
Comment (bguedas) [inline SDK6UserGuide/appendix/virtualDevice.rst:53]:
```suggestion
- the ``S3/`` folder, that contains the Simulator, HIL and Foundation Libraries specific to the Simulator 
```

File: SDK6UserGuide/appendix/virtualDevice.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+   |   |- veePort.properties
+   |   |- workbenchExtension*.jar
+
+The Virtual Device contains the Simulation part files of the VEE Port used to build it: 
+
+- the ``javaLibs/`` folder, that contains the Generic Foundation Libraries.
+
+- the ``linker/`` folder, that contains the Linker jar files. This folder is not embedded in the Virtual Device if an Architecture ``8.0.0`` is used.  
+
+- the ``MICROJVM/javaLibs/`` folder, that contains the Foundation Libraries specific to the MicroEJ Core Engine.
```
Comment (bguedas) [inline SDK6UserGuide/appendix/virtualDevice.rst:45]: I think that there should only be one entry for MICROJVM (as for S3). You can explain that it only contains files from VEE Port that are needed for building .fo files

File: SDK6UserGuide/appendix/virtualDevice.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+- the WPK files of System Applications that are extracted in their own folder, in the ``installed-applications/`` folder that is empty by default.
+
+.. note::
+
+   System Applications can only be embedded in a Multi-Sandboxed Virtual Device.
+   In case of a Mono-Sandboxed Virtual Device, the ``installed-applications/`` folder is always empty.
+
+- The ``options/application.properties`` that contains the Application properties and the properties of the VEE Port used to build the Virtual Device.
+
+- The ``scripts/init-vd/vd-init.xml`` script that allows to enable or not the Virtual Device. If the Virtual Device is not enabled, the Application main class specified by the user is launched on the VEE Port.
```
Comment (bguedas) [inline SDK6UserGuide/appendix/virtualDevice.rst:74]:
```suggestion
- the ``scripts/init-vd/vd-init.xml`` script, that allows to enable or not the Virtual Device. If the Virtual Device is not enabled, the Application main class specified by the user is launched on the VEE Port
```

File: SDK6UserGuide/appendix/virtualDevice.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+- the ``scripts/`` folder.
+
+- the ``tools/`` folder.
+
+- the ``workbenchExtension*.jar`` files.
+
+The following elements are also embedded in the Virtual Device:
+
+- a Kernel Application, whose WPK file is extracted in the ``main-application`` folder of the Virtual Device.
```
Comment (bguedas) [inline SDK6UserGuide/appendix/virtualDevice.rst:63]:
```suggestion
- a Kernel Application, whose WPK file is extracted in the ``main-application`` folder of the Virtual Device
```

File: SDK6UserGuide/buildVirtualDevice.rst [hunk 7]
```diff
@@ -0,0 +1,120 @@
+.. _sdk_6_build_virtual_device:
+
+Build a Virtual Device
+======================
+
+
```
Comment (bguedas) [inline SDK6UserGuide/buildVirtualDevice.rst:6]: empty line

File: SDK6UserGuide/buildVirtualDevice.rst [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+In order to build the Virtual Device of an Application, the SDK provides the Gradle ``buildVirtualDevice`` task.
+Refer to the :ref:`section.virtual.device` page for more information about the Virtual Device.
+
+The prerequisites to use the ``buildVirtualDevice`` task are:
+
+- The property ``applicationMainClass`` must be defined in the ``microej`` configuration block of the Gradle build file of the project.
+  It must be set to the Full Qualified Name of the Application main class, for example::
```
Comment (bguedas) [inline SDK6UserGuide/buildVirtualDevice.rst:13]:
```suggestion
  It must be set to the fully qualified name of the Application main class, for example::
```

File: SDK6UserGuide/buildVirtualDevice.rst [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+   - "veePort1"
+   - "veePort2"
+ 
+To build a Virtual Device, you must select the VEE Port to use by setting the ``veePort`` property to the 
+:ref:`unique name <sdk_6_vee_port_unique_name>` of the VEE Port when calling 
+the ``buildVirtualDevice`` task::
+
+   ./gradlew buildVirtualDevice -PveePort="veePortName"
+
+If you want to add the property in IntelliJ IDEA :
```
Comment (bguedas) [inline SDK6UserGuide/buildVirtualDevice.rst:62]:
```suggestion
If you want to add the property in IntelliJ IDEA: 
```

**Review by bguedas — CHANGES_REQUESTED** [review]

File: SDK6UserGuide/buildVirtualDevice.rst (hunk 7 above)
Comment (denisbedin) [reply SDK6UserGuide/buildVirtualDevice.rst]: Fixed.

File: SDK6UserGuide/appendix/virtualDevice.rst (hunk 2 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/virtualDevice.rst:36]: I fixed the bulleted lists.

File: SDK6UserGuide/appendix/virtualDevice.rst (hunk 4 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/virtualDevice.rst]: Done.

**Pull request merged**